### PR TITLE
Bl 301 fe as a user i want to know that my application successfully sent

### PR DESCRIPTION
**Pull request by farhaannishtar**
## Description
[Jira Ticket](https://bloomtechlabs.atlassian.net/jira/software/c/projects/BL/boards/10?modal=detail&selectedIssue=BL-301&search=elij)

We developed a landing page for the user to be directed to once they have submitted their application. The motivation for this change is to clearly inform the aspiring mentees / mentors that their application has been successfully sent and is in the process of being reviewed. 
There were no dependencies required for this change. 

## Loom Video
https://www.loom.com/share/5170e09778834b85a3d0a77e7745cc9a## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have removed unnecessary comments/console logs from my code
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] No duplicate code left within changed files
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

**Review by jettyblue — APPROVED** [review]
Brilliant! Diggin the application success page and code looks good. Also love the addition of the user's first name, nice personal touch.

File: src/components/pages/RoleSignup/Applications/AppSuccess.js
Comment (nathancai97) [inline src/components/pages/RoleSignup/Applications/AppSuccess.js:29]: I would change the wording, as this implies that the applicant will get an email of their approval within a time frame, rather than an email regarding if their application will be approved or not.

**Review by nathancai97 — CHANGES_REQUESTED** [review]

File: src/components/pages/RoleSignup/Applications/AppSuccess.js
Comment (ElijahHopkin) [reply src/components/pages/RoleSignup/Applications/AppSuccess.js:29]: Did you mean the text in green implies a time frame? One of the reasons we changed it was to make it less time-bound than the red text. Just making sure we're talking about the same lines.

Comment (nathancai97) [reply src/components/pages/RoleSignup/Applications/AppSuccess.js:29]: Yeah, I'm referring to the green line. It just refers that the applicant will receive an email when they're approved(unless I'm mistaken and Underdog Devs approves all applicants), not if they'll be approved or not.

Comment (ElijahHopkin) [reply src/components/pages/RoleSignup/Applications/AppSuccess.js]: Thanks for catching that. We just changed **when** to **if**.
let us know if you think anything else should be changed for that.

**Review by nathancai97 — DISMISSED** [review]
This landing page looks clean! The updated message, the inclusion of the applicant's name and the redirect button back to home is a great touch for the user experience. Awesome job!

File: src/components/pages/Dashboard/Dashboard.js
Comment (Johnjx) [inline src/components/pages/Dashboard/Dashboard.js:90]: Is there a reason for this commented-out code to remain?

File: src/index.js
Comment (Johnjx) [inline src/index.js:121]: Why has redirect been disabled for this route? Also, as per the ticket, is _/apply/success_ used as the default landing page if the user has an account and has not yet been approved? Please provide color if I am missing details.

File: src/components/pages/RoleSignup/Applications/Mentor.js
Comment (Johnjx) [inline src/components/pages/RoleSignup/Applications/Mentor.js:60]: Good job adding functionality to an existing function!

File: src/components/pages/RoleSignup/Applications/AppSuccess.js
Comment (Johnjx) [inline src/components/pages/RoleSignup/Applications/AppSuccess.js:43]: Nice state map and action creator grab. _userProfile_ prop does not appear to be in use. If not intended for use please remove.

File: src/components/pages/RoleSignup/Applications/AppSuccess.js
Comment (Johnjx) [inline src/components/pages/RoleSignup/Applications/AppSuccess.js:18]: I think that an overarching code comment may prove useful here to quickly explain what this component is responsible for and the reasoning for the input of the dependency array _isAuthenticated_. What would be the result if this component were to re-render on the Authentication status change? Bear in mind I may be missing some insight into the entire workings of the codebase!

**Review by Johnjx — COMMENTED** [review]
Great job succinctly achieving basic ticket functionality. I would like to see a little more context added overall. Perhaps the description could have included a link to a design file outlining the user flow. Also, commits could include more info such as 'connected X to Y for Z purpose' or as the user completes 'Z'.

File: src/components/pages/Dashboard/Dashboard.js
Comment (farhaannishtar) [reply src/components/pages/Dashboard/Dashboard.js:90]: Great catch. This console.log was showing in the dev console. I believe it was accidentally left there, so I commented it out and pushed it. We'll have Paul take a look and see what he thinks.

Comment (Johnjx) [reply src/components/pages/Dashboard/Dashboard.js:90]: Sounds good.

File: src/index.js
Comment (ElijahHopkin) [reply src/index.js:121]: Great question--We discussed this in our last product review meeting, and Ryan suggested the following flow for user onboarding/approval.
1. User Applies and lands on this page. At this point there is no login access given
2. one of the following:
a) super-admin will most likely approve them and they will then receive a link to create sign on info OR
b) it is possible that the super-admin will reject them. (rare--but possible). They will be informed --but never have sign on information given them.

This flow negates the description in the ticket you're referring to. It prevents us having to create users  that might need to be deleted later. Does that make sense?

Comment (ElijahHopkin) [reply src/index.js:121]: the redirect is also a good point. We can clean that up.

File: src/components/pages/Dashboard/Dashboard.js
Comment (ElijahHopkin) [reply src/components/pages/Dashboard/Dashboard.js:90]: This console log precedes this ticket--so we hadn't tackled it--but we should probably delete it. I'm curious what Paul has to say.

File: src/components/pages/RoleSignup/Applications/AppSuccess.js
Comment (MelodyLayne) [inline src/components/pages/RoleSignup/Applications/AppSuccess.js:33]: I like this use of the button as a redirect rather than timing the page out to go back to the main page. The only consideration I would make here is stylistic in that it might be more accessible to have the button filled in rather than outlined. But I also know that the Ant styling has been tricky so I wouldn't blame y'all for not wanting to open that can of worms.

**Review by MelodyLayne — APPROVED** [review]
Overall I think this is a really nicely done update. My only feedback was left in line with the code. Nice work y'all!

File: src/index.js
Comment (Johnjx) [reply src/index.js:121]: Great @ElijahHopkin, thanks for providing context on that!

**Review by zdravoj — DISMISSED** [review]
Good changes. I like your addition of the user's first name in the "application received" page, that's a nice personal touch.

**Review by MelodyLayne — DISMISSED** [review]

**Review by jettyblue — APPROVED** [review]
Landing page looks nice and clean, nice work!

**Review by nathancai97 — APPROVED** [review]
Landing page looks great, love the home button redirecting the user!

**Review by MelodyLayne — APPROVED** [review]

**Review by techlando — APPROVED** [review]
nicely done adding functionality.

**Review by brimjustin — APPROVED** [review]
Good job changing this to something that gives the user a better understanding of the process that is happening!

File: src/components/pages/Dashboard/Dashboard.js
Comment (paulstgermain) [reply src/components/pages/Dashboard/Dashboard.js:90]: I actually tend to agree - we should all take responsibility for cleaning up our codebases when we see something like ugly commented-out code! Everyone pitching in is how huge companies operate and keep things running smoothly. 😉

**Review by MelodyLayne — APPROVED** [review]

**Review by MelodyLayne — DISMISSED** [review]

**Review by brimjustin — APPROVED** [review]
Effective changes made and good job changing this to something that gives the user a better understanding of the process that is happening!

File: src/components/pages/RoleSignup/Applications/Mentee.js
Comment (miguelaledesma) [inline src/components/pages/RoleSignup/Applications/Mentee.js:54]: I love the use of useHistory, super easy to just send the user to the success page

**Review by nathancai97 — APPROVED** [review]

**Review by miguelaledesma — DISMISSED** [review]
Nice work with the landing page! Glad you all were able to make this work, Its nice and clean, i saw that there was some console.logs but looks like you guys are going to take care of that or already have. Good Work ya'll!

**Review by Nicholas-R-Payne — DISMISSED** [review]
Landing page looks good! Good idea adding home button to redirect users after success!

**Comment by ElijahHopkin** [comment]
sorry all. the removal of okta required that we create a new signup link to access the application process.
We've utilized the blue signup button from the navbar.

**Review by albert-szabo — APPROVED** [review]
Good job applying changes in a thoughtful way that is unambiguous and informative to the user! In particular, the use of the new useHistory feature and de-structuring to keep things concise is a nice touch.

File: src/components/pages/RoleSignup/Applications/README.md
Comment (Johnjx) [inline src/components/pages/RoleSignup/Applications/README.md:12]: Great README, I love this strategy to help explain the current state to devs without having to find a video or the person who made the changes.

**Review by Johnjx — APPROVED** [review]
Great job guys. It certainly takes effort to keep up with the migrating DB structure and maintain a working state of the application, and you are making this transition easier for the rest of us!

File: src/components/pages/Dashboard/Dashboard.js
Comment (ryan-hamblin) [reply src/components/pages/Dashboard/Dashboard.js:90]: to me this seems like the commented out log is still in the code.

File: src/components/pages/RoleSignup/Applications/Mentee.js
Comment (ryan-hamblin) [inline src/components/pages/RoleSignup/Applications/Mentee.js:57]: a simple technique that could be used prior to pushing the user is to wait a few seconds using a setTimeout to push user ensuring that the 200 response comes back from the userPost.

Comment (ryan-hamblin) [reply src/components/pages/RoleSignup/Applications/Mentee.js:57]: something that occurred us during our PR Critiques session was the idea that this particular dispatch() could be a promise returned. If that be the case we might be able to `thenify` the method and control through the promise return --> routing user to successfully page or setting the error from the server in state and showing that as necessary to the user.

Comment (ryan-hamblin) [reply src/components/pages/RoleSignup/Applications/Mentee.js:57]: https://blog.jscrambler.com/async-dispatch-chaining-with-redux-thunk

You may find a better to do this, but there are a few examples to reflect on here.

File: src/components/pages/RoleSignup/Applications/Mentor.js
Comment (ryan-hamblin) [inline src/components/pages/RoleSignup/Applications/Mentor.js:62]: belete it

**Review by ryan-hamblin — APPROVED** [review]
solid work. I love that are making progress on the Auth and protected route flow. 

Let's consider the items discussed around handling those errors moving forward.

**Pull request merged**
